test/e2e/specs/editor/various/editor-modes.spec.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Editing modes (visual/HTML)', () => {
+	test.beforeEach( async ( { admin, editor } ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Hello world!' },
+		} );
+	} );
+
+	test( 'should switch between visual and HTML modes', async ( {
+		editor,
+	} ) => {
+		const paragraphBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+
+		// This block should be in "visual" mode by default.
+		await expect( paragraphBlock ).toHaveClass( /rich-text/ );
+
+		// Change editing mode from "Visual" to "HTML".
+		await editor.clickBlockOptionsMenuItem( 'Edit as HTML' );
+
+		// Wait for the block to be converted to HTML editing mode.
+		await expect( paragraphBlock.getByRole( 'textbox' ) ).toBeVisible();
+
+		// Change editing mode from "HTML" back to "Visual".
+		await editor.clickBlockOptionsMenuItem( 'Edit visually' );
+
+		// This block should be in "visual" mode by default.
+		await expect( paragraphBlock ).toHaveClass( /rich-text/ );
```

Comment:
```suggestion
		// This block should be in "visual" mode.
		await expect( paragraphBlock ).toHaveClass( /rich-text/ );
```

Nit: It seems like it doesn't matter what the default mode is here, what do you think?